File: backend/app/models/task.py
from pydantic import BaseModel
from typing import Optional
from datetime import datetime


class Task(BaseModel):
    id: str
    title: str
    description: str
    tags: list[str]
    status: str
    created_at: str
    updated_at: str


class CreateTaskInput(BaseModel):
    title: str
    description: str
    tags: list[str] = []

File: frontend/src/types/task.ts
export type TaskStatus = 'pending' | 'analyzing' | 'completed' | 'failed';

export interface Task {
  id: string;
  title: string;
  description: string;
  tags: string[];
  status: TaskStatus;
  createdAt: string;
  updatedAt: string;
}

export interface CreateTaskInput {
  title: string;
  description: string;
  tags: string[];
}

export interface UpdateTaskInput {
  title?: string;
  description?: string;
  tags?: string[];
  status?: TaskStatus;
}
